src/main/java/com/example/java/maven/ticketVendingMachine/CoinStorage.java

```diff
@@ -44,43 +50,34 @@ public int getValueOfCoinsInStorage() {
         return valueOfCoinsInStorage;
     }
 
-    public static int getRequiredNumberOfCoinsWithGivenValue(int oddMoney, int coinValue) {
-        return Math.floorDiv(oddMoney, coinValue);
-    }
-
-    public static boolean areRequiredCoinsAvailableInCoinStorage(int requiredNumberOfCoins, int coinValue, CoinStorage mainCoinStorage) {
-        if (getNumberOfCoinsWithGivenValueInGivenCoinStorage(coinValue, mainCoinStorage) >= requiredNumberOfCoins) {
+    public static boolean areRequiredCoinsAvailableInCoinStorage(int requiredNumberOfCoins, Coin coin, CoinStorage mainCoinStorage) {
+        if (mainCoinStorage.coins.get(coin) >= requiredNumberOfCoins) {
```

Comment:
just return: `mainCoinStorage.coins.get(coin) >= requiredNumberOfCoins` if statement is unnecessary

Reply:
This method is about to get deleted due to improvements in giving back the change - for now it's not used, but I'll leave it temporarily in suggested form.

Reply:
Method deleted.